src/DayNine/Smoke2.java

```diff
@@ -27,18 +27,16 @@ public void printLowPointCoordsToHash() {
     }
 
     public void checkBasinSize(int[][] lowPointCoordArray, int[][] heightMap) { //todo
-        Iterator<Point> i = lowPoints.iterator();
-        while (i.hasNext()) {
-            seenPoints.removeAll(seenPoints);
+        for (Point point : lowPoints) {
+            seenPoints.clear();
             int m = 1;
-            Point point = i.next();
             seenPoints.add(point);
             Set<Point> seenThisIteration = new HashSet<>();
             Iterator<Point> j = seenThisIteration.iterator();
             boolean done = false;
             if (j.hasNext()) {
                 done = false;
-                point =j.next();
+                point = j.next();
```

Comment:
Ik wil deze ook verforren, maar de loop is nog niet af zo te zien, dus ik weet nog niet hoe.